### ISSUE TITLE
[Terminal >Set tab color]: Screen Reader is not announcing role as 'On/Off' for Selected/Non-selected Colors button.

### ISSUE THREAD

**Issue by v-rpundir** [microsoft/terminal#12043]
### Windows Terminal version

1.12.3472.0

### Windows build number

10.0.22504.1010

### Other Software

**Test Environment:**
OS: Windows 11 Version Dev (OS Build 22504.1010)
App: Windows Terminal Preview
Screen Reader: Narrator
Tool: Accessibility Insights for Windows Version 1.1.1741.1

### Steps to reproduce

**Repro Steps:**
1. Open Windows Terminal.
2. Open Settings page using 'Ctr+,'.
3. Activate 'Actions' tab and then Add a new actions 'Set tab color' with shortcut key 'Ctrl + Shift + c'.
4. Open Narrator using 'Win + Ctrl + Enter' keys.
5. navigate inside Terminal window.
6. Press 'Ctrl + Shift + c' keys, then navigate to colors control inside Set tab color pop-up using tab key.
7. Observe the issues.

**User Experience:**
Screen Reader users UX will not be good as they will get full control info as role properties not announced to them.

**Guideline Reference:**
https://www.w3.org/WAI/WCAG21/Understanding/name-role-value

**Attachments:**
[Screen Reader is not announcing role as 'OnOff' for SelectedNon-selected Colors button..zip](https://github.com/microsoft/terminal/files/7779518/Screen.Reader.is.not.announcing.role.as.OnOff.for.SelectedNon-selected.Colors.button.zip)

### Expected Behavior

Screen Reader should announce role as 'On/Off' for Selected/Non-selected Colors button.
i.e. For the Selected 'Steel blue' button Screen Reader should announce 'Steel blue button On' and for non-selected 'Medium Sea Green' button Screen Reader should announce 'Medium Sea Green button Off'.

### Actual Behavior

Screen Reader is not announcing role as 'On/Off' for Selected/Non-selected Colors button.
i.e. For Selected 'Steel blue' button Screen Reader is only announcing 'Steel blue button' and for non-selected 'Medium Sea Green' button Screne Reader is only announcing 'Medium Sea Green button'.

**Observation:** Issue repro with NVDA and JAWS Screen Reader.

**Comment by carlos-zamora** [comment]
I'm not sure about this one. @DHowett and @zadjii-msft for thoughts if I'm missing anything.

So, the short version of the bug is:
1. set tab color to "Steel blue"
2. open tab color picker _again_

Expected: "Steel blue" button should be "on", but other color buttons should be "off"
Actual: all buttons are just buttons. No state of "on"/"off".

This is by design. Setting the color is a one-directional scenario. By that, I mean that selecting a color is a "set" operation with no relationship with the existing color of the tab. This is important to retain because there are several other ways to set the tab color (i.e. VT escape sequences, theming, profile setting, etc...). Using the color picker overrides the existing color.

Thus, I'm closing this as "by design".

**Comment by DHowett** [comment]
This sounds right to me. Tab color is not _selected_... each button performs an immediate action, and there is no toggle aspect to it -- by design, because there is no capacity for us to track "the application set its own tab color, we must represent it differently in the UI",
